src/libs/ajax/User.ts

```diff
@@ -74,17 +94,44 @@ export interface SetTerraUserProfileRequest {
   interestInTerra?: string;
 }
 
-export const makeSetUserProfileRequest = (terraUserProfile: TerraUserProfile): SetTerraUserProfileRequest => {
+export const generateAPIBodyForCreateUserProfile = (
+  request: CreateTerraUserProfileRequest
+): OrchestrationUpsertTerraUserProfileRequest => {
+  return generateAPIBodyForUpdateUserProfile(request);
+};
+
+export const generateAPIBodyForUpdateUserProfile = (
+  request: UpdateTerraUserProfileRequest
+): OrchestrationUpsertTerraUserProfileRequest => {
   return {
-    firstName: terraUserProfile.firstName ?? 'N/A',
-    lastName: terraUserProfile.lastName ?? 'N/A',
-    title: terraUserProfile.title ?? 'N/A',
-    institute: terraUserProfile.institute ?? 'N/A',
-    programLocationCity: terraUserProfile.programLocationCity ?? 'N/A',
-    programLocationState: terraUserProfile.programLocationState ?? 'N/A',
-    programLocationCountry: terraUserProfile.programLocationCountry ?? 'N/A',
-    department: terraUserProfile.department,
-    interestInTerra: terraUserProfile.interestInTerra,
+    // TODO: this is the logic to look at when writing the tests
+    // first name and last name are REQUIRED for this request, and they are populated from the oidc token claims,
+    // so they should not be undefined
+    firstName: request.firstName!,
+    lastName: request.lastName!,
+
+    // contact email is NOT REQUIRED for this request, but it is populated from the oidc token claims,
+    // so it should not be undefined
```

Comment:
I went in double checked in orch (https://github.com/broadinstitute/firecloud-orchestration/blob/abb4597ba4031bbf659290b0a36ef1134572b6c5/src/main/scala/org/broadinstitute/dsde/firecloud/model/Profile.scala#L22).

Contact email is not required in orch, but it is a required field as part of the form in`Register.ts`. The default value of contactEmail is email (from the token claims) in that form.

In `PersonalInfo`, `contactEmail` could be technically undefined (although under normal circumstances it should be `''`), so maybe this could be changed to: 

`contactEmail: request.contactEmail`